contracts/optimism/connectors/connext/main.sol

```diff
@@ -0,0 +1,67 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @title Connext.
+ * @dev Cross chain bridge.
+ */
+
+import { TokenInterface, MemoryInterface } from "../../common/interfaces.sol";
+import { Stores } from "../../common/stores.sol";
+
+import "./interface.sol";
+import "./helpers.sol";
+import "./events.sol";
+
+abstract contract ConnextResolver is Helpers {
+	/**
+	 * @dev Call xcall on Connext.
+	 * @notice Call xcall on Connext.
+	 * @param params XCallParams struct.
+	 * @param getId ID to retrieve amount from last spell.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function xcall(XCallParams memory params, uint256 getId, uint256 setId)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amount = getUint(getId, params.amount);
+		TokenInterface tokenContract = TokenInterface(params.asset);
+		bool isNative = params.asset == ethAddr;
+
+		if (isNative) {
+			_amount = _amount == uint256(-1) ? sub(address(this).balance, params.relayerFee) : _amount;
```

Comment:
```suggestion
			_amount = _amount == uint256(-1) ? address(this).balance : _amount;
```

Since it would be a delegate call through DSA, I believe we don't have to keep a relayer gas fee margin for native token.